clients/src/test/java/org/apache/kafka/test/MockSelector.java

```diff
@@ -100,14 +112,20 @@ public void completeReceive(NetworkReceive receive) {
         this.completedReceives.add(receive);
     }
 
+    public void delayedReceive(DelayedReceive receive) {
+        this.delayedReceives.add(receive);
+    }
+
     @Override
     public List<String> disconnected() {
         return disconnected;
     }
 
     @Override
     public List<String> connected() {
-        return connected;
+        List<String> currentConnected = new ArrayList<>(connected);
+        connected.clear();
```

Comment:
It seems a bit inconsistent that `connected` clears the internal list while other methods like `disconnected` don't. Maybe we should call `clear()` from `poll`?